tests/structures/test_generator.py

```diff
@@ -231,28 +247,246 @@ def gen():
                 pass
             """)
 
-    @expectedFailure
     def test_generator_yield_expr_return(self):
         self.assertCodeExecution("""
             def gen():
                 return (yield)
 
             g = gen()
             g.send(None)
-            g.send(1)  # for some reason the generator keeps returning value
-            g.send(100) # without raising StopIteration error
+            try:
+                g.send(1)
+            except StopIteration:
+                pass
+            """)
+
+    @expectedFailure
+    def test_generator_yield_try_finally_special_case(self):
+        """Output is 'finally' followed by 'Hello World'
+        due to the way CPython handles garbage collection."""
+        self.assertCodeExecution("""
+            def gen():
+                try:
+                    yield 'Hello World'
+                finally:
+                    print('finally')
+
+            print(next(gen()))
+            """)
+
+    def test_generator_throw_on_starting(self):
+        self.assertCodeExecution("""
+            def gen():
+                yield "Hello World"
+
+            g = gen()
+            try:
+                g.throw(ZeroDivisionError)
+            except ZeroDivisionError:
+                pass
```

Comment:
The VOC test suite is based on comparing output, so having tests that "pass" and do nothing in a branch is potentially risky. Ideally, you want to print something that confirms this branch was actually executed (and was executed in the right order); you also want to have print statements in places that *can't* be reached (e.g., after the throw statement) to confirm that code that *should* be dead is actually dead.

Reply:
Resolved.